File: example/e2e-demo/pytest/src/discount.py
"""Discount calculation logic — the pytest track's own small module,
independent of the TS cart module (no cross-language imports)."""


def apply_discount(total: float, percent: float) -> float:
    """Applies a percentage discount to a total.

    Args:
        total: The pre-discount amount.
        percent: The discount percentage (0-100).

    Returns:
        The discounted amount, rounded to 2 decimal places.
    """
    if not 0 <= percent <= 100:
        raise ValueError("percent must be between 0 and 100")
    return round(total * (1 - percent / 100), 2)

File: example/e2e-demo/go/discount.go
// Package e2edemo mirrors the pytest discount module for the Go track — its
// own small module, independent of the TS cart module (no cross-language imports).
package e2edemo

import (
	"errors"
	"math"
)

// ApplyDiscount applies a percentage discount to a total, rounded to 2 decimal places.
func ApplyDiscount(total float64, percent float64) (float64, error) {
	if percent < 0 || percent > 100 {
		return 0, errors.New("percent must be between 0 and 100")
	}
	discounted := total * (1 - percent/100)
	return math.Round(discounted*100) / 100, nil
}
